pkg/sql/backfill.go

```diff
@@ -1216,9 +1223,16 @@ func (sc *SchemaChanger) validateForwardIndexes(
 					ie.tcModifier = nil
 				}()
 
+				filter := ""
+				// If the index is a partial index the predicate must be added
+				// as a filter to the query to force scanning the index.
+				if idx.IsPartial() {
+					filter = fmt.Sprintf(`WHERE %s`, idx.Predicate)
+				}
+
 				row, err := ie.QueryRowEx(ctx, "verify-idx-count", txn,
 					sqlbase.InternalExecutorSessionDataOverride{},
-					fmt.Sprintf(`SELECT count(1) FROM [%d AS t]@[%d]`, tableDesc.ID, idx.ID))
+					fmt.Sprintf(`SELECT count(1) FROM [%d AS t]@[%d]%s`, tableDesc.ID, idx.ID, filter))
```

Comment:
do you need a space here after the `[%d]`?

Reply:
I would expect it to be a syntax error but it's not.

```
root@127.0.0.1:53107/defaultdb> select * from [52 as t]@[2]WHERE b > 0;
  a | b
----+----
(0 rows)
```

Regardless, I've add a space at the beginning of the predicate string to make query this more idiomatic.